### ISSUE TITLE
Add support for .NET Core

### ISSUE THREAD

**Issue by batzen** [MeikTranel/NXPorts#8]
I would like to contribute the code needed to support .NET Core.
This would require:
- special code for `CallingConvetion`
- support in dnlib (already created a PR there)

**Comment by MeikTranel** [comment]
Can you link the PR in dnlib?

**Comment by MeikTranel** [comment]
Also: Is it possible to implement the custom changes required for .NET Core support using dnlib's public api? I don't want to put the owner of dnlib in a position where they have to support something solely for NXPorts.

**Comment by batzen** [comment]
My PR got merged and the changes were not specific to NXPorts.
I just added the ability to overwrite the name of the dll being used in the ImportDirectory.
So yes, it will be possible to just use the public API to add support for .NET Core.

To be honest, i just started to try a different approach of injecting snoop into the running process as ijwhost does, currently, not work with self contained apps and snoop also has to work in that scenario.
So i am not sure if i still need dllexports in c#.
If i drop the dllexport approach i will tell you so and point you to the sources needed to support .NET Core in NXPorts as i won't contribute it in that case. Sorry in advance.

**Comment by batzen** [comment]
And the PR was https://github.com/0xd4d/dnlib/pull/281

**Comment by MeikTranel** [comment]
That's okay. I'll put this under the future milestone. I'm personally not hugely interested in .NET Core support.. It'd be cool, but as of right now .NET Core does not support this scenario at all. Any issues in regards to hosting .net core through PInvoke would move into the consumers own risk. Meanwhile i'll try to continue to push the .NET team to add this scenario into the next releases as a first class citizen.
I think C API calls are a very strong usecase to support interop between platforms looking at the bigger picture.

**Comment by Sewer56** [comment]
> continue to push the .NET team to add this scenario into the next releases as a first class citizen.

Yes please. There are many interesting use cases that are yet to be opened up due to a lack of a C interface. In my opinion the most interesting ones are plugin development for native applications and interop in the `Linux/OSX/Unix` world. 

Sure it's possible to use the official way by running a .NET Core host in a native process with hostfxr  and delegating native exports to the .NET runtime but it's really a pain in the back side to do so.
